model/src/main/kotlin/OrtResult.kt

```diff
@@ -93,6 +94,13 @@ data class OrtResult(
             evaluator = null,
             labels = emptyMap()
         )
+
+        /**
+         * Load an [OrtResult] from the given [file]. This function deserializes the result from the specified input
+         * file and ensures that it has been properly initialized. It therefore should be always used rather than
+         * doing the deserialization manually.
+         */
+        fun load(file: File): OrtResult = file.readValue<OrtResult>().withResolvedScopes()
```

Comment:
I think this could also have been solved by providing a custom deserializer for `OrtResult` which calls the function after deserialization.